pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1556,6 +1594,66 @@ func (d *awsDisk) describeVolume() (*ec2.Volume, error) {
 	return volumes[0], nil
 }
 
+func (d *awsDisk) describeVolumeModification() (*ec2.VolumeModification, error) {
+	volumeID := d.awsID
+	request := &ec2.DescribeVolumesModificationsInput{
+		VolumeIds: []*string{volumeID.awsString()},
+	}
+	volumeMods, err := d.ec2.DescribeVolumeModifications(request)
+
+	if err != nil {
+		return nil, fmt.Errorf("Error describing volume modification %s with %v", volumeID, err)
+	}
+
+	if len(volumeMods) == 0 {
+		return nil, fmt.Errorf("No volume modifications found for %s", volumeID)
+	}
+	lastIndex := len(volumeMods) - 1
```

Comment:
Instead of checking the latest, we could match by operation start time, it looks like.  That may be worse though... thoughts?

Reply:
There will be only one volume modification object for a volume at a time. Volume can't be modified if there is already a modification in progress. Also, it appears that there is a per-volume quota of 1 modification per 6 hour, so I think we should be okay.